libr/core/core.c

```diff
@@ -2205,7 +2208,38 @@ static char *get_comments_cb(void *user, ut64 addr) {
 	return r_core_anal_get_comments ((RCore *)user, addr);
 }
 
-static void cb_event_handler(REvent *ev, REventType event_type, void *data) {
+R_IPI void spaces_list(RSpaces *sp, int mode) {
+	RBIter it;
+	RSpace *s;
+	bool first = true;
+	const RSpace *cur = r_spaces_current (sp);
+	if (mode == 'j') {
+		r_cons_printf ("[");
+	}
+	r_rbtree_foreach (sp->spaces, it, s, RSpace, rb) {
+		int count = r_spaces_count (sp, s->name);
+		if (mode == 'j') {
+			r_cons_printf ("%s{\"name\":\"%s\"%s,\"count\":%d}",
+				       !first? ",": "", s->name,
+				       cur == s? ",\"selected\":true": "",
+				       count);
+		} else if (mode == '*') {
+			r_cons_printf ("%s %s\n", sp->name, s->name);
+		} else {
+			r_cons_printf ("%d %c %s\n", count, cur == s? '*': '.',
+				       s->name);
+		}
+		first = false;
+	}
+	if (mode == '*' && r_spaces_current (sp)) {
+		r_cons_printf ("%s %s # current\n", sp->name, r_spaces_current_name (sp));
+	}
+	if (mode == 'j') {
+		r_cons_printf ("]\n");
+	}
+}
+
+static void cb_event_handler(REvent *ev, int event_type, void *user, void *data) {
```

Comment:
Completely unrelated to this PR, but @radare this is wrong. You should create separate functions for each event_type, instead of using a switch/case. It's much much cleaner. We all know how we ended up with switch/cases for commands :D